### PR TITLE
Make REST settings context agnostic.

### PR DESCRIPTION
**Pull request by Hug0-Drelon**
## What?
Fixes https://github.com/polylang/polylang-pro/issues/2376

## How?
- Remove `context` parameter from options schema.
- Remove checks for context in `Settings`.
- Adapt tests accordingly.

### CODE REVIEW

**Comment by Screenfeed** [comment]
*”Cough”* https://github.com/polylang/polylang/pull/1569#discussion_r1843903317

**Comment by Hug0-Drelon** [comment]
> _”Cough”_ [#1569 (comment)](https://github.com/polylang/polylang/pull/1569#discussion_r1843903317)

It was the misunderstanding of the behavior of the method. I thought it was adding `edit` to the schema for *our* properties that didn't have that set. No properties passed along a request.
Moreover, the behavior was not tested at all and passed under the radar.

Also, we may need to rollback about this context, see https://github.com/polylang/polylang-pro/issues/2376#issuecomment-2518637200

File: tests/phpunit/tests/test-rest-settings.php [hunk 1]
```diff
@@ -163,6 +163,15 @@ public function test_unknown_method() {
 		$this->assertSame( 404, $response->get_status() );
 	}
 
+	public function test_get_settings_with_no_context() {
```
Comment (Screenfeed) [inline tests/phpunit/tests/test-rest-settings.php:166]: This could go in `test_wrong_context()`.

**Review by Screenfeed — CHANGES_REQUESTED** [review]

**Comment by Screenfeed** [comment]
>Moreover, the behavior was not tested at all and passed under the radar.

It was already tested (that's why we have 4 tests failing) and didn't passed under the radar (since it was expected).

File: tests/phpunit/tests/test-rest-settings.php (hunk 1 above)
Comment (Hug0-Drelon) [reply tests/phpunit/tests/test-rest-settings.php]: Not applicable anymore.

**Comment by Hug0-Drelon** [comment]
The PR changes its purpose, now we don't care about `context` anymore.

**Pull request merged**